src/Symfony/Component/DependencyInjection/ParameterBag/EnvPlaceholderParameterBag.php

```diff
@@ -71,6 +71,12 @@ public function getEnvPlaceholders()
      */
     public function mergeEnvPlaceholders(self $bag)
     {
-        $this->envPlaceholders = array_merge_recursive($this->envPlaceholders, $bag->getEnvPlaceholders());
+        $newPlaceholders = $bag->getEnvPlaceholders();
```

Comment:
@nicolas-grekas I was trying to make a test for this but I need your input.

param bag A has placeholders:
`'DB_USER' => ['env_DB_USER_1']`

param bag B has placeholders:
`'DB_USER' => ['env_DB_USER_1'], 'DB_HOST_1' => ['env_DB_HOST_1]`

After the merge I would expect param bag A to have placeholders:
`'DB_USER' => ['env_DB_USER_1'], 'DB_HOST_1' => ['env_DB_HOST_1]`

right?

Now lets make things a little more interesting with param bag C which has a different unique identifier for one of the placeholders:

`'DB_USER' => ['env_DB_USER_2']`

So now we go and merge that into param bag A (after merging B). Do you expect:

A) `'DB_USER' => ['env_DB_USER_2'], 'DB_HOST_1' => ['env_DB_HOST_1]`

or 

B) `'DB_USER' => ['env_DB_USER_1', 'env_DB_USER_2'], 'DB_HOST_1' => ['env_DB_HOST_1]`

If you're expecting B) like I am then I'll happily make your changes (which work as expected) once I reintroduce the `$this->envPlaceholders[$env][$placeholder] = $placeholder;` from above. If I left it without it your suggestion wouldn't work. The changes on my branch as they are now work in either case. I'm happy with both, using `+=` looks nice and neat.

Just let me know the decision and i'll push the changes and the new test.

Sorry for the delay, just want to be sure I get this right this time.